src/lib/mod.rs

```diff
@@ -49,58 +54,72 @@ impl Transaction {
 
 impl TransactionType {
     pub fn compute_and_hash(&self) -> Result<String> {
-        let mut hasher = Sha256::new();
-
         match self {
             TransactionType::FunctionExecution {
-                program: program_str,
+                program: _,
                 function,
-                program_name: _,
-                enable_trace: execute_trace,
+                program_name,
+                enable_trace: _
             } => {
-                let program = Program::from_reader(program_str.as_bytes(), None)?;
-                let mut vm = VirtualMachine::new(*execute_trace);
-
-                let mut cairo_runner = CairoRunner::new(&program, "all", false)?;
-
-                let mut hint_processor = BuiltinHintProcessor::new_empty();
-
-                let entrypoint = program
-                    .identifiers
-                    .get(&format!("__main__.{function}"))
-                    .and_then(|x| x.pc)
-                    .context("Error geting entrypoint function")?;
-
-                cairo_runner.initialize_builtins(&mut vm)?;
-                cairo_runner.initialize_segments(&mut vm, None);
-
-                cairo_runner.run_from_entrypoint(
-                    entrypoint,
-                    &[
-                        &MaybeRelocatable::from(2).into(),
-                        &MaybeRelocatable::from((2, 0)).into(),
-                    ],
-                    false,
-                    &mut vm,
-                    &mut hint_processor,
-                )?;
-                cairo_runner.relocate(&mut vm).unwrap();
-
-                let trace = cairo_runner.relocated_trace;
-
-                match trace {
-                    Some(trace) => {
-                        for reg in trace {
-                            hasher.update(serde_json::to_string(&reg)?);
-                        }
-                    }
-                    None => info!("Trace not enabled, not executing/hashing"),
-                }
+
+                let general_config = StarknetGeneralConfig::default();
+
+                let tx_execution_context =
+                    TransactionExecutionContext::create_for_testing(
+                        Address(0.into()),
+                        10,
+                        0.into(),
+                        general_config.invoke_tx_max_n_steps(),
+                        1,
+                    );
+
+                let contract_address = Address(1111.into());
+                let class_hash = [1; 32];
+                let contract_class =
+                    ContractClass::try_from(PathBuf::from("programs").join(program_name))
```

Comment:
`ContractClass` only allowed me to pass a path not the contents of it , I could do the following by using [starknet-api](https://github.com/lambdaclass/starknet-api), but unsure if it is the right way or if we even have to fix this now

```rust
let raw_contract_class: starknet_rs::state::ContractClass = serde_json::from_str(program)?;
let contract_class = ContractClass::from(raw_contract_class);
```

Reply:
Yeah I think this is fine as a first step. If we wanted to make it compatible though, you could create a temp file and have it read from there. Otherwise if the ABCI does not have the program on disk it won't be able to execute it.